Let users customize the flags passed to xcpretty

gfontenot opened the pull request:
This adds two variables that can be set to customize xcpretty:
- `xcpretty_flags` will be applied to all xcpretty operations. By
  default, this is set to `--color`.
- `xcpretty_testing_flags` will only be applied to test actions. This
  is empty by default.

fixes #7

gfontenot committed:
Let users customize the flags passed to xcpretty

This adds two variables that can be set to customize xcpretty:

 - `xcpretty_flags` will be applied to all xcpretty operations. By
   default, this is set to `--color`.
 - `xcpretty_testing_flags` will only be applied to test actions. This
   is empty by default.

gfontenot commented:
ping @gabebw @christoomey @keith any comments here? Is this the right way to go about this?

keith commented:
Code looks good to me. I'm not sure differentiating them is necessary though, but I haven't spent a lot of time looking at home xcpretty formats just the build part with different test flags.

gabebw commented:
This looks reasonable.

You may want to prefix the global variables with xcodebuild, like `g:xcodebuild_xcpretty_flags` etc, to prevent conflicts (unlikely) and make it clear where the variable is being used.

gabebw commented on plugin/xcodebuild.vim:
```diff
@@ -41,8 +49,17 @@ endfunction
+  let xcpretty = s:xcpretty()
+  if empty(xcpretty)
+    return ''
+  endif
```
How about an if/else?

``` vim
if empty(xcpretty)
  return ''
else
  return xcpretty . ' ' . g:xcpretty_testing_flags
endif
```

gfontenot commented:
> I'm not sure differentiating them is necessary though

When passing `--test` during a build action, it silences the actual build output. So you end up with just the return state. Seems like not what we want, at least not by default.

gfontenot committed:
Prefer single quotes over double quotes

This is just a personal style preference.

gfontenot committed:
Use script functions instead of resetting globals

Instead of checking to see if a global exists and then setting it, I
feel like it would be nicer to use internal functions to achieve the
same result.

gfontenot commented:
Updated to take comments into account. I've also switched _away_ from setting the global if it doesn't exist, and instead am using an internal function for the same effect. I don't know if that's recommended, but it seems cleaner to me for some reason.
